src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.Ecdh.cs

```diff
@@ -11,8 +11,100 @@ internal static partial class Interop
-        internal static partial SafeEvpPKeyCtxHandle EvpPKeyCtxCreate(SafeEvpPKeyHandle pkey, SafeEvpPKeyHandle peerkey, out uint secretLength);
+        private static partial SafeEvpPKeyCtxHandle EvpPKeyCtxCreate(SafeEvpPKeyHandle pkey, IntPtr extraHandle, SafeEvpPKeyHandle peerkey, out uint secretLength);
+
+        internal static SafeEvpPKeyCtxHandle EvpPKeyCtxCreate(SafeEvpPKeyHandle pkey, SafeEvpPKeyHandle peerkey, out uint secretLength)
+            => EvpPKeyCtxCreate(pkey, pkey.ExtraHandle, peerkey, out secretLength);
+
+        [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpPKeyCtxConfigureForECDSASign")]
+        private static partial int EvpPKeyCtxConfigureForECDSASignCore(SafeEvpPKeyCtxHandle ctx);
+
+        internal static void EvpPKeyCtxConfigureForECDSASign(SafeEvpPKeyCtxHandle ctx)
+        {
+            Debug.Assert(ctx != null);
+            Debug.Assert(!ctx.IsInvalid);
+
+            if (EvpPKeyCtxConfigureForECDSASignCore(ctx) != 1)
+            {
+                throw CreateOpenSslCryptographicException();
+            }
+        }
+
+        [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpPKeyCtxConfigureForECDSAVerify")]
+        private static partial int EvpPKeyCtxConfigureForECDSAVerifyCore(SafeEvpPKeyCtxHandle ctx);
+
+        internal static void EvpPKeyCtxConfigureForECDSAVerify(SafeEvpPKeyCtxHandle ctx)
+        {
+            Debug.Assert(ctx != null);
+            Debug.Assert(!ctx.IsInvalid);
+
+            if (EvpPKeyCtxConfigureForECDSAVerifyCore(ctx) != 1)
+            {
+                throw CreateOpenSslCryptographicException();
+            }
+        }
+
+        [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpPKeyCtxSignHash")]
+        private static unsafe partial int EvpPKeyCtxSignHash(SafeEvpPKeyCtxHandle ctx, byte* hash, int hashLen, byte* destination, ref int destinationLen);
+
+        internal static unsafe bool TryEvpPKeyCtxSignHash(SafeEvpPKeyCtxHandle ctx, ReadOnlySpan<byte> hash, Span<byte> destination, out int bytesWritten)
+        {
+            Debug.Assert(ctx != null);
+            Debug.Assert(!ctx.IsInvalid);
+
+            if (hash.Length == 0 || destination.Length == 0)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            bytesWritten = destination.Length;
+            ref byte hashRef = ref MemoryMarshal.GetReference(hash);
+            ref byte destRef = ref MemoryMarshal.GetReference(destination);
+            fixed (byte* hashPtr = &hashRef)
+            fixed (byte* destPtr = &destRef)
+            {
+                return EvpPKeyCtxSignHash(ctx, hashPtr, hash.Length, destPtr, ref bytesWritten) == 1;
+            }
+        }
+
+        internal static unsafe bool TryEvpPKeyCtxSignatureSize(SafeEvpPKeyCtxHandle ctx, ReadOnlySpan<byte> hash, out int bytesWritten)
```

Comment:
Is this ever needed as a "Try"?

If yes, the output parameter shouldn't be `bytesWritten`, since they weren't written to.  Probably `signatureSize`.

And why does this take a hash input?

I think this all goes away based on my other feedback...